src/proto/tagpost.proto

```diff
@@ -7,17 +7,68 @@ option java_package = "com.google.tagpost";
 option java_outer_classname = "TagPostProto";
 
 service TagpostService {
+  // dummy rpc service
   rpc FetchMessage(FetchMessageRequest) returns (FetchMessageResponse) {}
+  // fetch all threads by primary tag
+  rpc FetchThreadsByTag(FetchThreadsByTagRequest) returns (FetchThreadsByTagResponse) {}
+  // fetch all comments under one thread
+  rpc FetchCommentsUnderThread(FetchCommentsUnderThreadRequest) returns (FetchCommentsUnderThreadResponse) {}
+  // fetch all comments by a set of tags
+  rpc FetchCommentsByTags(FetchCommentsByTagsRequest) returns (FetchCommentsByTagsResponse) {}
+}
+
+message Tag {
+  string tag_name = 1;
+}
+
+message TagpostThread {
+  int64 thread_id = 1;
+  string primary_tag = 2;
+}
+
+message Comment {
+  int64 comment_id = 1;
+  int64 thread_id = 2;
+  string comment_content = 3;
+  string timestamp = 4;
```

Comment:
Use this proto for timestamps:
https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#timestamp